File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 0
        self.get_logger().info("Hello Ros2")
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self.counter_ +=1
        self.get_logger().info("Hello " +str(self.counter_))

def main(args=None):
    rclpy.init(args=args) 
    ##node = Node("py_test")
    ##node.get_logger().info("Hello Ros2")
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "/rclcpp/rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Nodes
{
public:
MyNode(): Node("cpp_test"), counter_(0)
{
    RCLCPP_INFO(this->get_logger(), "Hello Cpp Node")

    timer_= this.create_wall_timer(std::chrono::seconds(1), 
    std::bind(&MyNode::timerCallback, this));
}
private:
void timerCallback()
{
    counter_++; 
    RCLCPP_INFO(this->get_logger(), "Hello %d",counter_)

}
rclcpp::TimerBase::SharedPtr timer;
int counter_;
} 

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);//initialize ros2 communication
    //auto node = std::make_shared>rclcpp::Node>("cpp_test");
    auto node = std::make_shared<MyNode>();//create sharepointer to the node
    RCLCPP_INFO(node->get_logger(), "Hello Cpp Node")
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
} 